src/decision/src/LidarDecision.cpp

```diff
@@ -131,11 +131,26 @@ LidarDecision::LidarDecision(int argc, char **argv, std::string node_name) {
         twist.angular.y = 0;
         twist.angular.z = 0;
 
-        if (obstacle.getAvgDistance() < danger_distance && abs(obstacle.getAvgAngle()) < danger_angle) {
-            // TODO: Improve the algorithm here to at least use some sort of scale  (linear or otherwise) dependent on obstacle distance and angle
-            // Check if we should be turning left or right
-            int left_or_right = (obstacle.getAvgAngle() > 0) ? -1 : 1;
-            twist.angular.z = angular_vel_multiplier * left_or_right;
+        if (obstacle.getMinDistance() <= danger_distance && obstacle.getAvgAngle() <= danger_angle &&
+            obstacle.getAvgAngle() >= -danger_angle) {
+            float minDistance = obstacle.getMinDistance();
+            twist.linear.x = sqrt(minDistance);
+            // Calculate angular z based on how aligned obstacle is to the robot
+            float angle_dep_score = abs(cos(obstacle.getAvgAngle()));
```

Comment:
I think it would be fine just to name there `distance_score` and `angle_score` respectively, adding `dep` is a little confusing, and isn't really needed to convey the **intent** that the scores are *dependent* on distance and angle respectively.